core/vm/evm.go

```diff
@@ -356,8 +386,9 @@ func (evm *EVM) StaticCall(caller ContractRef, addr common.Address, input []byte
 	// future scenarios
 	evm.StateDB.AddBalance(addr, big0)
 
+	isExternal := evm.isExternalContract(addr)
 	// Invoke tracer hooks that signal entering/exiting a call frame
-	if evm.Config.Debug {
+	if evm.Config.Debug && !isExternal {
```

Comment:
I'm leaving the comment here because I cannot on the lines I would. I see on line 218 this condition `if evm.Config.Debug && !isExternal` before a `Tracer.CaptureStart` and `Tracer.CaptureStart` invocation. The same is on line 391 for `Tracer.CaptureEnter` and `Tracer.CaptureExit`. I wonder if we need to add the external check also on lines 290. I also saw the same condition on lines, 476 and 521, but they are in the `create` and `delegateCall` function, which won't ever be executed from an external contract, so they should be fine

Reply:
As you have noted, there is no need to check for external contract for methods that cannot be executed by an external contract. This applies to `create` because it can only be executed by the EVM, to `delegateCall` and `CallCode` (that is the function with line 290) because they are not supported by external contracts and the callback is not invoked. This way we can still have the tracing for these methods.